tests/approaches/test_oracle_approach.py

```diff
@@ -388,15 +399,19 @@ def test_cluttered_table_get_gt_nsrts(place_version=False):
             assert env.action_space.contains(dump_action.arr)
             env.simulate(state, dump_action)  # never raises EnvironmentFailure
         else:
-            place3_nsrt = place_nsrt.ground([can3])
+            print("hihi again")
+            place1_nsrt = place_nsrt.ground([can1])
             with pytest.raises(AssertionError):
-                place_nsrt.ground([can3, can1])
-            place_option = place3_nsrt.sample_option(state, task.goal, rng)
+                place_nsrt.ground([can0, can1])
+
+            print("hihi again 2!")
+            place_option = place1_nsrt.sample_option(state, task.goal, rng)
+            print("hihi again 3!")
             place_action = place_option.policy(state)
             assert env.action_space.contains(place_action.arr)
             try:
                 env.simulate(state, place_action)
-            except utils.EnvironmentFailure as e:
+            except utils.EnvironmentFailure as e:  # pragma: no cover
```

Comment:
let's try to see if we can cover this case (and remove the pragma: no cover). the EnvironmentFailure will get raised when there's a collision upon placing. if you want, you can manually create the `place_action` to force such a collision (by extracting the objects and state from `state`, seeing where the goal object's pose is). or, you might be able to just change the code above to something like:
```
train_tasks = env.get_train_tasks()
...
if i < len(train_tasks) / 2:
            utils.reset_config(
                {"cluttered_table_place_goal_conditioned_sampling": False})
        else:
            utils.reset_config(
                {"cluttered_table_place_goal_conditioned_sampling": True})
```
because I imagine that collisions are more likely with the non-goal conditioned sampler.

Reply:
ok, I was following some other patterns in the code where I noticed we didn't necessarily cover exceptions. through some careful if statements this case  is now covered...